c_src/quicer_stream.c

```diff
@@ -316,10 +316,19 @@ async_start_stream2(ErlNifEnv *env,
     }
 
   //
-  // note, ctx is not shared yet, thus no locking is needed.
+  // note, s_ctx is not shared yet, thus no locking is needed.
   //
   QuicerStreamCTX *s_ctx = init_s_ctx();
+
+  enif_mutex_lock(c_ctx->lock);
+  if (c_ctx->is_closed)
+    {
+      enif_mutex_unlock(c_ctx->lock);
```

Comment:
`destroy_s_ctx(s_ctx)` is missing?

Reply:
nice catch, I should move this block ahead.